tests/components/history/test_websocket_api.py

```diff
@@ -1584,3 +1584,233 @@ async def test_overflow_queue(
+    hass.states.async_set("sensor.one", "on", attributes={"any": "attr"})
+    await async_recorder_block_till_done(hass)
+    hass.states.async_set("sensor.two", "off", attributes={"any": "attr"})
+    await async_recorder_block_till_done(hass)
+    hass.states.async_set("switch.excluded", "off", attributes={"any": "again"})
+    await async_wait_recording_done(hass)
+
+    await async_wait_recording_done(hass)
+
+    client = await hass_ws_client()
+
+    await client.send_json(
+        {
+            "id": 1,
+            "type": "history/stream",
+            "start_time": now.isoformat(),
+            "include_start_time_state": True,
+            "significant_changes_only": False,
+            "no_attributes": True,
+        }
+    )
+    response = await client.receive_json()
+    assert response["success"]
+    response = await client.receive_json()
+    assert "sensor.one" in str(response)
+    assert "sensor.two" in str(response)
+
+    await client.send_json(
+        {
+            "id": 2,
+            "type": "history/stream",
+            "start_time": now.isoformat(),
+            "entity_ids": ["sensor.one"],
+            "include_start_time_state": True,
+            "significant_changes_only": False,
+            "no_attributes": True,
+        }
+    )
+    response = await client.receive_json()
+    assert response["success"]
+    response = await client.receive_json()
+    assert "sensor.one" in str(response)
+
+    await client.send_json(
+        {
+            "id": 3,
+            "type": "history/stream",
+            "start_time": now.isoformat(),
+            "entity_ids": ["sensor.one", "sensor.two"],
+            "include_start_time_state": True,
+            "significant_changes_only": False,
+            "no_attributes": True,
+        }
+    )
+    response = await client.receive_json()
+    assert response["success"]
+    response = await client.receive_json()
+    assert "sensor.one" in str(response)
+    assert "sensor.two" in str(response)
+
+    await client.send_json(
+        {
+            "id": 4,
+            "type": "history/stream",
+            "start_time": now.isoformat(),
+            "entity_ids": ["sens!or.one", "two"],
+            "include_start_time_state": True,
+            "significant_changes_only": False,
+            "no_attributes": True,
+        }
+    )
+    response = await client.receive_json()
+    assert response["success"] is False
+    assert "invalid_entity_ids" in str(response)
+
+    await client.send_json(
+        {
+            "id": 5,
+            "type": "history/stream",
+            "start_time": now.isoformat(),
+            "entity_ids": ["sensor.one", "sensortwo."],
+            "include_start_time_state": True,
+            "significant_changes_only": False,
+            "no_attributes": True,
+        }
+    )
+    response = await client.receive_json()
+    assert response["success"] is False
+    assert "invalid_entity_ids" in str(response)
```

Comment:
Since the response is structured data we should assert its in the right place in the response instead of stringifying the response